### PR TITLE
chore: support telemetry test

### PR DESCRIPTION
**Pull request by chagong**
chore: support telemetry test

To support telemetry UITEST we need to send telemetry into a log file instead of sending to AI. The test need to set a env variable  (TEAMSFX_TELEMETRY_TEST) to enable it.

image

https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/10676875

### CODE REVIEW

File: packages/vscode-extension/src/commonlib/telemetry.ts [hunk 1]
```diff
@@ -19,19 +31,53 @@ import { Correlator } from "@microsoft/teamsfx-core";
...
   constructor(key: string, extensionVersion: string, extensionId: string) {
     super(async () => await this.reporter.dispose());
     this.reporter = new Reporter(extensionId, extensionVersion, key, true);
     this.extVersion = getPackageVersion(extensionVersion);
+    const logFile = path.join(os.homedir(), `.${ConfigFolderName}`, TelemetryTestLoggerFile);
+    configure({
+      appenders: { everything: { type: "file", filename: logFile } },
+      categories: { default: { appenders: ["everything"], level: "debug" } },
+    });
+    this.logger = getLogger("TelemTest");
```
Comment (swatDong) [inline packages/vscode-extension/src/commonlib/telemetry.ts:48]: Is the log file created here or on the first log generated?

**Review by XiaofuHuang — APPROVED** [review]

File: packages/vscode-extension/src/commonlib/telemetry.ts (hunk 1 above)
Comment (chagong) [reply packages/vscode-extension/src/commonlib/telemetry.ts:48]: created at 'getLogger' call.

Comment (chagong) [reply packages/vscode-extension/src/commonlib/telemetry.ts:48]: oh, was wrong, it's at 'configure' call.

**Comment by HuihuiWu-Microsoft** [comment]
Is it a good practice to include testing code in production? What about stub methods of this.reporter before test cases?

File: packages/vscode-extension/src/commonlib/telemetry.ts (hunk 1 above)
Comment (swatDong) [reply packages/vscode-extension/src/commonlib/telemetry.ts:48]: So how about move `configure` and `getLogger` in if condition? For normal user we don't want to generate the test log file.

```
if (TelemetryTest) {
   configure...
   getLogger...
}
```

**Comment by chagong** [comment]
> Is it a good practice to include testing code in production? What about stub methods of this.reporter before test cases?

It's for UITEST. It's not like UT that we can mock some objects or functions. it has to be in production code.

**Review by swatDong — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Pull request merged**